### PR TITLE
[DNM] Mineral floors

### PR DESCRIPTION
**Pull request by unid15**
Turf sprites borrowed from tg

Will polish tomorrow

### CODE REVIEW

File: code/game/turfs/simulated/floor.dm [hunk 1]
```diff
@@ -255,6 +263,41 @@ turf/simulated/floor/proc/update_icon()
...
+/turf/simulated/floor/is_plasma_floor()
+	if(istype(floor_tile,/obj/item/stack/tile/mineral/plasma))
+		return 1
+	return 0
+
+/turf/simulated/floor/is_diamond_floor()
+	if(istype(floor_tile,/obj/item/stack/tile/mineral/diamond))
+		return 1
+	return 0
+
```
Comment (PJB3005) [inline code/game/turfs/simulated/floor.dm]: Can't you make these into defines?

File: code/game/turfs/simulated/mineral.dm [hunk 2]
```diff
@@ -0,0 +1,128 @@
+/turf/simulated/floor/mineral
+	name = "mineral floor"
+	icon_state = ""
+
+
+
+/turf/simulated/floor/mineral/New()
+	..()
+
```
Comment (PJB3005) [inline code/game/turfs/simulated/mineral.dm]: Is this even useful?

Comment (clusterfack) [inline code/game/turfs/simulated/mineral.dm]: No this does nothing

Comment (PJB3005) [inline code/game/turfs/simulated/mineral.dm]: Wait, it does do something:

Block return values

honk

Comment (clusterfack) [inline code/game/turfs/simulated/mineral.dm]: Only if new has a return value and it isn't null

**Comment by clusterfack** [comment]
turf/simulated/floors have a variable called "mineral" which is defined as metal, perhaps you should be changing that variable to the respective mineral and then setting icon_state and such from that?

**Comment by Unfit** [comment]
now I've fixed most of the issues, made use of the material var in floor turfs etc
also added phazon floor tiles which make you go fast (like you have hyperzine in your system)

things to do
1) fix conflicts
2) add icons for burnt/broken tiles

**Comment by Unfit** [comment]
WHAT IS HAPPENING

**Pull request closed without merging**